src/Agent.Listener/SelfUpdater.cs

```diff
@@ -147,6 +149,36 @@ private async Task<bool> UpdateNeeded(string targetVersion, CancellationToken to
             Trace.Info($"Current running agent version is {BuildConstants.AgentPackage.Version}");
             PackageVersion agentVersion = new PackageVersion(BuildConstants.AgentPackage.Version);
 
+            //Checking if current system support .NET 6 agent
+            if (agentVersion.Major == 2 && serverVersion.Major == 3)
+            {
+                Trace.Verbose("Checking if your system supports .NET 6");
+
+                try
+                {
+                    OS[] supportedSystems = GetSupportedSystemsNet6();
+
+                    string systemId = PlatformUtil.SystemId;
+                    OSVersion systemVersion = PlatformUtil.SystemVersion;
+
+                    Trace.Verbose($"The system you are running on: '{systemId}' ({systemVersion})");
+
+                    if (!supportedSystems.Any((system) => system.Equals(systemId, systemVersion)))
+                    {
+                        Trace.Warning($"It doesn't look like the system supports .NET 6, skipping update");
+                        return false;
+                    }
+                    else
+                    {
+                        Trace.Verbose("The system persists in the list of systems supporting .NET 6");
+                    }
+                }
+                catch (Exception ex)
+                {
+                    Trace.Error($"Error has occurred while checking if system supports .NET 6: {ex} ");
```

Comment:
It seems this exception throws when reading/parsing net6.json file, can we write more specific error message?

Reply:
Exception can be related to `net6.json` file and to determining of current system Id and Version. I think we can left generic catch here.